collections/_posts/2024-11-13-AlphaFold2.md

```diff
@@ -0,0 +1,208 @@
+
+One of the key ideas of this transformer is to incorporate clever inductive biases inside the operations performed by the model. As the number of dimensions of the inputs tensors is large -- they can be seen as embedded 2D images --, instead of performing cross-attention on the flattened sequences of tokens, the MSA is modified via row-wise and column-wise self attention operations. Moreover, the MSA only contains feature information, without any information about the structure of the protein. To incorporate such knowledge into the MSA representation, the authors use a modified version of the row-wise self-attenetion where they add inside the softmax operation biases coming from the pair representation tensor (Figure 6). Note that this is only done for the row-wise operations and not the the column-wise ones as it does not really make sense to add this kind of structural information, specific to each protein, to a cross-attention of the information of one amino-acid between severel species (column-wise self-attention is basically the same as Figure 6 without the lower part of the schema).
+
+<div style="text-align:center">
+<img src="/collections/images/AlphaFold/row_wise.jpg" width=800></div>
+<p style="text-align: center;font-style:italic">Figure 6. Details of the row-wise gated self-attention with pair bias.</p>
+
+&nbsp;
+
+The Evoformer also updates the pair representation, using element-wise addition with the updated MSA tensor followed by self-attention blocks. The paper also introduces very import novelty in these attention blocks. Their goal is to update the embedding of the structural representation of the molecule. The generic self-attention does not impose specific bias on the representation. However, if we think about the amino acid chain as a concatenation of small triangles (formed by the carbons and nitrogen of the backbone chain), one property that should satisfy every triplet of point is the triangular inequality : when considering a triangle, a length of a side can not be greater than the sum of the lengths of the two other sides of the triangle. Such property, for the author, is key when one try to predict the shape of the 3D molecule. We can not directly impose such condition in the network (plus we are dealing here with embedded positions, not strictly the Euclidian position of the atom for now), but the pair representation are updated with a modified self-attention whose goal is to *mimic the spirit of the triangular inequality*. For an edge $$e_{ij}$$, we will consider all the possible triangle with edges $$e_{ij}$$, $$e_{ik}$$ and $$e_{ik}$$ (for $$k  \neq i,j$$) and the edge $$e_{ij}$$ will receive an from a combination of the two other edges of each of these triangles (Figure 7). The pair representation updates are composed of four similar self-attention, respectively using "outgoing edges", "incoming edges", "starting node" or "ending node".
```

Comment:
Seems to be a missing word in _will receive an (?) from a combination_, and I'm not able to guess what it's supposed to be.

Reply:
the missing word was upadte but I slightly changed the sentence to avoid a repetition